eventloop/eventloop.go

```diff
@@ -24,18 +24,35 @@ func GetGlobalEventLoop() *EventLoop {
 	return GlobalEventLoop
 }
 
-func (e *EventLoop) Await(currentP *Promise) interface{} {
+func (e *EventLoop) Await(currentP *Promise) (interface{}, error) {
 	defer currentP.Done()
 	currentP.RegisterHandler()
-	val := <-currentP.rev
-	return val
+	for {
+		select {
+		case err := <-currentP.errChan:
+			return nil, err
+		case rev := <-currentP.rev:
+			return rev, nil
+		}
+	}
+
 }
 
 func (e *EventLoop) Async(fn func() (interface{}, error)) *Promise {
 	resultChan := make(chan interface{})
 	errChan := make(chan error)
 	p := e.newPromise(resultChan, errChan)
 	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				switch x := r.(type) {
+				case error:
+					p.errChan <- x
+				default:
+					p.errChan <- fmt.Errorf(`unknown error: %v`, x)
+				}
+			}
+		}()
```

Comment:
Remove this recovery till we can think of a better approach to handle panic in the `Async` func